MSAL/src/MSALResult.m

```diff
@@ -55,15 +61,18 @@ + (MSALResult *)resultWithAccessToken:(NSString *)accessToken
     return result;
 }
 
-+ (MSALResult *)resultWithAccessTokenItem:(MSALAccessTokenCacheItem *)cacheItem
++ (MSALResult *)resultWithAccessToken:(MSIDAccessToken *)accessToken
 {
-    return [self resultWithAccessToken:cacheItem.accessToken
-                             expiresOn:cacheItem.expiresOn
-                              tenantId:cacheItem.tenantId
-                                  user:cacheItem.user
-                               idToken:cacheItem.rawIdToken
-                              uniqueId:cacheItem.uniqueId
-                                scopes:[cacheItem.scope array]];
+    __auto_type idToken = [[MSIDAADV2IdTokenWrapper alloc] initWithRawIdToken:accessToken.idToken];
+    MSALUser *user = [[MSALUser alloc] initWithIdToken:idToken
+                                            clientInfo:accessToken.clientInfo environment:accessToken.authority.msidHostWithPortIfNecessary];
+    return [self resultWithAccessToken:accessToken.accessToken
+                             expiresOn:accessToken.expiresOn
+                              tenantId:accessToken.authority.msidTenant
+                                  user:user
+                               idToken:accessToken.idToken
+                              uniqueId:idToken.objectId
```

Comment:
is MSAL uniqueId objectID or it also has a fallback to subject?

Reply:
Checked android, that use subject as a fallback. Changed to have same logic in our code.